routing/index_graph_starter.hpp

```diff
@@ -24,9 +24,47 @@ namespace routing
 class IndexGraphStarter final
 {
 public:
-  // AStarAlgorithm types aliases:
-  using TVertexType = Joint::Id;
-  using TEdgeType = JointEdge;
+  // IndexGraph is a G = (V, E), where V is a Joint, and E is a pair
+  // of joints. But this space is too restrictive, as it's not
+  // possible to add penalties for left turns, U-turns, restrictions
+  // and so on. Therefore, we run A* algorithm on a different graph G'
+  // = (V', E'), where each vertex from V' is a pair of vertices
+  // (previous vertex, current vertex) from V. Following laws hold:
+  //
+  // 1. (s, s) in V', where s is a start joint.
+  //
+  // 2. (t, t) in V', where t is a finish joint.
+  //
+  // 3. If there is an edge (u, v) in E and edge (v, w) in E, then
+  // there is an edge from (u, v) to (v, w) in E', and the weight of
+  // this edge is the same as the weight of (v, w) from E.
+  //
+  // 4. If there is an edge (s, u) in E then there is an edge from (s,
+  // s) to (s, u) in E' and the weight of this edge is the same as the
+  // weigth of (s, u) from E.
+  //
+  // 5. If there is an edge (u, t) in E then there is an edge from (u,
+  // t) to (t, t) in E' of zero length.
+  //
+  // 6. (s, s) from V' and (t, t) from V' are start and finish
+  // vertices correspondingly.
+  //
+  // As one can see, vertices in the new space are actually edges from
+  // the original space.
+  using TVertexType = pair<Joint::Id, Joint::Id>;
+
+  struct TEdgeType
+  {
+  public:
+    TEdgeType(TVertexType const & target, double weight) : m_target(target), m_weight(weight) {}
+
+    TVertexType GetTarget() const { return m_target; }
+    double GetWeight() const { return m_weight; }
+
+  private:
+    TVertexType m_target;
+    double m_weight;
```

Comment:
Let's make them const. As far as I see you don't change them after construction.